src/Database/Type/EnumType.php

```diff
@@ -111,13 +111,8 @@ public function toPHP(mixed $value, Driver $driver): ?BackedEnum
             return null;
         }
 
-        if (get_debug_type($value) !== $this->backingType) {
-            throw new InvalidArgumentException(sprintf(
-                'Cannot convert value of type %s to %s with type %s',
-                get_debug_type($value),
-                $this->enumClassName,
-                $this->backingType
-            ));
+        if ($this->backingType === 'int' && is_string($value)) {
```

Comment:
Should this be `ctype_digit()`? to be stricter on our type juggling?

Reply:
Enums can have negative values so that check won't work on its own.

Reply:
We could use `filter_var()` instead then. My concern is a value like `'3 bears'` ending up here and being cast to `3` without an error.

Reply:
Sure. It will be a little more costly, but there probably isn't a large number of enum types in data.

Reply:
Done. I don't know how much more it actually costs since were doing a string to int cast before which has to parse.

Reply:
fyi, we don't perform this validation in the regular IntegerType.

Reply:
I don't think overly strict validation is necessary here.

Reply:
> I don't think overly strict validation is necessary here.

I thought being more strict with enums would help avoid hiding errors in application code. I don't see the extra validation as a must and we can take it out if you think it is only adding overhead.